### PR TITLE
ENH: Add closed generator to randint

### PR DESCRIPTION
**Pull request by bashtage**
Add closed option to randint to simplify some cases

### CODE REVIEW

**Comment by mattip** [comment]
Can we hold off on this until after we first merge randomgen to numpy? I would hate for the merge to get bogged down in discussions of the merit of this enhancement. Or was there consensus around adding this kwarg to randint?

**Comment by bashtage** [comment]
Im happy to wait.

**Comment by bashtage** [comment]
xref numpy/numpy#13164

File: numpy/random/bounded_integers.pyx.in [hunk 1]
```diff
@@ -50,13 +50,21 @@ cdef object _rand_{{nptype}}_broadcast(np.ndarray low, np.ndarray high, object s
...
+        low_high_comp = np.greater
+    else:
+        high_comp = np.greater
+        low_high_comp = np.greater_equal
+
+    if np.any(high_comp(high_arr, {{ub}})):
         raise ValueError('high is out of bounds for {{nptype}}')
-    if np.any(np.greater_equal(low_arr, high_arr)):
+    if np.any(low_high_comp(low_arr, high_arr)):
         raise ValueError('low >= high')
```
Comment (eric-wieser) [inline numpy/random/bounded_integers.pyx.in]: This error message doesn't match the comparator

Comment (bashtage) [reply numpy/random/bounded_integers.pyx.in]: I think that string probably needs to be dynamic and change depending on whether closed or not.

Comment (bashtage) [reply numpy/random/bounded_integers.pyx.in]: low=high is ok for closed, but only low<high for open.

**Comment by bashtage** [comment]
I pushed some small changes in response to the feedback above.  This is needed since the closed interval is in the API through `integers(low, high, endpoint=True)`, which is just a refactor once this is in.

**Comment by mattip** [comment]
In light of the discussion, I would like to put this in now. Any objections?

**Comment by bashtage** [comment]
It failed on my Travis, FWIW.  It is failing due to changes in pytest. 

https://travis-ci.org/bashtage/numpy/builds/531943929

File: numpy/random/bounded_integers.pyx.in (hunk 1 above)
Comment (bashtage) [reply numpy/random/bounded_integers.pyx.in]: This has been fixed.

**Comment by mattip** [comment]
> FWIW. It is failing due to changes in pytest.

numpy/numpy#13550 should fix that

**Pull request merged**